field.go

```diff
@@ -410,6 +410,14 @@ func Inline(val zapcore.ObjectMarshaler) Field {
 	}
 }
 
+type anyFieldC[T any] func(string, T) Field
+
+func (f anyFieldC[T]) Any(key string, val any) Field {
```

Comment:
Should this be private? Does it matter if it's public?

Reply:
It doesn't matter if it's public or private here.
An advantage of `Any` is that it's not going to conflict with `any`, the built-in type.
I also try to document the "public" interface of internal types by making them public like this.
(So someone using anyFieldC knows they can use Any, but they can't use someInternalStateManagement.)